### ISSUE TITLE
several issues around bus flow creation

### ISSUE THREAD

**Issue by rjmaris** [eclipse/sumo#10073]
Upon creation of a bus flow from scratch, the following single issues arise:
1. (**Fixed**) When a bus flow is created, it gets an ID. When I want to add stops (A mode) on already (newly created) bus stops - not from OSM import - the dropdown list "Parent element" is empty, despite several parents having been created. I only can enter the ID in the text field by hand.
2. (**Fixed**) The inspect view shows me that the vehs per hour is still at its default of 1800 (despite setting it to 6 upon creation). I had to change it in inspect mode.
3. Assumption: The route is automatically set up according to the bus stops that will be defined by the user. As such, there would be no need to add additional edges between start and end. This is correct, but only partially. When I add a new stop@bus stop outside of the shortest route, and then hit F5 (compute), the orange route line is automatically updated to follow the bus stop! When I start the simulation, the bus did follow the shortest route in an instance created before new verification on writing this. HENCE: The behaviour seems not exactly reproducible. Therefore I asses further review before making this point "hard".
4.  It's easy to add additional bus stops assigned to a flow by multiple clicks (of course that does not make sense). In order to check this, I just view the demand xml file for a quick review if I did not accidentally double stops.
5. If possible, the colour of an actual bus stop should not already be red (due to assignment to any bus id) when a new stop@bus stop is created for a certain bus id. The same should apply if a route is highlighted via inspect - only those red which are really assigned to the highlighted route - this allows a quick check if no stop is forgotten.

  
---
*If possible, upload an example that shows your problem.*

**SUMO-version:** 1.12

**operating system:** Linux Mint 20.4

List of sub-tickets for easier reference:
- #10074
- #10075
- #10076 
- #10077 
- #10078 
- #10079
- related: #10120

**Comment by namdre** [comment]
I can reproduce (1) and confirm it's a bug: #10074 Sometimes, I have a single element in the parent drop-down but it definitely doesn't show all eligible objects reliably.

**Comment by namdre** [comment]
I can also confirm (2) as a bug: #10075

**Comment by namdre** [comment]
A step-by-step description for reproducing (3) would be helpful. I suspect that the difference you are seing could be related to the ordering of stops (though netedit currently does not make it easy to verify the order #10076)

**Comment by namdre** [comment]
(4) should be handled via #10077 and #10078

**Comment by namdre** [comment]
(5) should be handled via #10077 and #10079

**Comment by namdre** [comment]
@rjmaris Thank your for reporting your findings!

**Comment by rjmaris** [comment]
Regarding (3) - Bus Stops, I can definitely confirm that in version 1.12 no quirks seem to occur. I'm not sure whether I did the first experiences using 1.11, because the upgrade was just around that time.

First of all, the experienced limit of max. 4 edges to specifiy for a vehicle flow seems to have been removed as of 1.12 (but I find no account of that in the Change Log). As such, there is no limit to exactly follow the intended route, when e.g. 10 stops at bus stops are on it. Hence the 'via' routing approach makes simply sense.
However, the job is simpler&quicker, when one practises the most common setup of start and end edge, and let the route be determined by clicks on bus stops (which must be executed as a sequence after route definition). Indeed this works reliably as of 1.12: Hitting F5 after a stop at bus stop is clicked does update the orange route line. Also the route has been simulated correcty.
Bottom lines:
- What would happen if the order of _stop@bus stop_ lines in the .xml file could change upon file save? - as remarked by namdre.
- If the intended bus route between two bus stops is not the shortest one, one would still need a "via" parameter to get it right.
